specify/specify7
# Add series tool in query builder

Fixes #2000 
Fixes #6166 

### Checklist

      and self-explanatory (or properly documented)

### Testing instructions

- Open the QB with CO table 
- Add catalog num field and some other field 
- [ ] verify series checkbox is present only when Cat num is one of the field 
- Check Series 
- [ ] verify Distinct is read only 
- Uncheck Series 
- [ ] Verify you can check Distinct and that Series become readonly 
- Uncheck Distinct and check Series 
- Run the query 
- [ ] Verify that all the records that have the same data for each field in your query and that have consecutive cat num are group together 
- [ ] Verify that the consecutive cat numbers are displayed as a range 

---
Tips: 
- Create a new CO with a determination and a preparation in a db with auto-numbering for cat num 
- Save 
- Use bulk Carry and create 5 identical copies 
- Create another new CO with all the exact same info as the previous one but skip one cat num 
- This will give you some record to test with series 

---
With Series 
![Screenshot 2025-02-03 at 8 50 57 AM](https://github.com/user-attachments/assets/9cc9d5f2-49f2-404a-babd-7b5664c52515)

Without Series
![Screenshot 2025-02-03 at 8 50 49 AM](https://github.com/user-attachments/assets/8be69707-8601-4dc0-992f-a1f5dbeaafb5)

NOTES: 
Sorting the column cat num is not available for series
